components/google_sheets/sources/common.js

```diff
@@ -97,8 +99,26 @@ module.exports = {
     getWorksheetIds() {
       throw new Error("getWorksheetIds is not implemented");
     },
+    processEvent() {
+      throw new Error("processEvent is not implemented");
+    },
     takeSheetSnapshot() {
       throw new Error("takeSheetSnapshot is not implemented");
     },
   },
+  async run(event) {
+    const isInitialized = this.db.get("isInitialized");
+    if (!isInitialized) {
+      await this.takeSheetSnapshot();
+      this.db.set("isInitialized", true);
+    }
+
+    const { headers } = event;
+    if (headers["x-goog-resource-state"] === "sync") {
```

Comment:
If the component is invoked by a timer, `headers` won't be defined and this will throw.

Reply:
Fixed.